idx/idx-api.php

```diff
@@ -531,47 +531,92 @@ public function saved_link_properties( $saved_link_id ) {
-		$listing_data = [];
-		// Property listings taken from $listing_data['data'].
-		$properties = [];
+		// Handle supplemental listings.
+		if ( 'supplemental' === $type ) {
+			// Pass 'featured' to get just the active supplemental listings.
+			return $this->get_client_supplementals( 'featured' );
+		}
+
+		$properties        = [];
+		$download_complete = false;
+
+		// Make initial API request for listings.
+		$listing_data = $this->idx_api( "$type?disclaimers=true", Initiate_Plugin::IDX_API_DEFAULT_VERSION, 'clients', array(), 7200, 'GET', true );
 
-		// Initial request.
-		$listing_data = $this->idx_api( $type . "?disclaimers=true&offset=$offset", Initiate_Plugin::IDX_API_DEFAULT_VERSION, 'clients', array(), 7200, 'GET', true );
-		// Assign returned listings to $properties.
 		if ( isset( $listing_data['data'] ) && is_array( $listing_data['data'] ) ) {
 			$properties = $listing_data['data'];
 		}
 
-		// Get rest of listings if there are more than 50.
-		if ( isset( $listing_data['total'] ) ) {
-			while ( ( $offset + $limit ) < $listing_data['total'] ) {
-				$offset = $offset + $limit;
-				$listing_data = $this->idx_api( $type . "?disclaimers=true&offset=$offset", Initiate_Plugin::IDX_API_DEFAULT_VERSION, 'clients', array(), 7200, 'GET', true );
-				$properties   = array_merge( $properties, $listing_data['data'] );
+		// Download remaining listings if available.
+		while ( ! $download_complete ) {
+			// Check if there is a next URL to request more listings.
+			if ( empty( $listing_data['next'] ) ) {
+				$download_complete = true;
+				continue;
+			}
+			// Explode $listing_data['next'] on '/clients/', index 1 of the resulting array will have the fragment needed to make the next API request.
+			$listing_data = $this->idx_api( explode( '/clients/', $listing_data['next'] )[1], Initiate_Plugin::IDX_API_DEFAULT_VERSION, 'clients', array(), 7200, 'GET', true );
+			// If $listing_data['data'] is an array, merge it with the existing listings/properties array.
+			if ( isset( $listing_data['data'] ) && is_array( $listing_data['data'] ) ) {
+				$properties = array_merge( $properties, $listing_data['data'] );
 			}
 		}
 
-		// Get any supplemental listings.
-		$supplemental_listings = $this->idx_api( 'supplemental', Initiate_Plugin::IDX_API_DEFAULT_VERSION, 'clients', array(), 7200, 'GET', true );
-
-		if ( is_array( $supplemental_listings ) ) {
-			$properties = array_merge( $properties, $supplemental_listings );
+		// Add supplemental listings to featured and soldpending types.
+		if ( 'featured' === $type || 'soldpending' === $type ) {
+			return array_merge( $properties, $this->get_client_supplementals( $type ) );
 		}
 
+		// Fallback, return $properties array.
 		return $properties;
 	}
 
+
+	/**
+	 * Get_Client_Supplementals function.
+	 * Helper function to gather supplemental listings.
+	 *
+	 * @access public
+	 * @param bool $active_only - active only supplemental listings flag.
```

Comment:
Mismatch between commented behavior and code, no comment param for `$status`, `$active_only` not used. See also lines 608 and 612